### PR TITLE
feat(scalasdk): generate action sources for Scala SDK

### PR DESCRIPTION
**Pull request by octonato**

### CODE REVIEW

File: codegen/java-gen/src/main/scala/com/lightbend/akkasls/codegen/java/ActionServiceSourceGenerator.scala [hunk 1]
```diff
@@ -73,14 +73,8 @@ object ActionServiceSourceGenerator {
     List(implSourcePath, interfaceSourcePath, providerSourcePath, handlerSourcePath)
   }
 
-  private def isUnary(cmd: ModelBuilder.Command): Boolean = !cmd.streamedInput && !cmd.streamedOutput
-  private def isStreamIn(cmd: ModelBuilder.Command): Boolean = cmd.streamedInput && !cmd.streamedOutput
-  private def isStreamOut(cmd: ModelBuilder.Command): Boolean = !cmd.streamedInput && cmd.streamedOutput
-  private def isStreamInOut(cmd: ModelBuilder.Command): Boolean = cmd.streamedInput && cmd.streamedOutput
-  private def hasStream(cmd: ModelBuilder.Command): Boolean = isStreamIn(cmd) || isStreamOut(cmd) || isStreamInOut(cmd)
```
Comment (octonato) [inline codegen/java-gen/src/main/scala/com/lightbend/akkasls/codegen/java/ActionServiceSourceGenerator.scala:80]: moved into Command as we need it now in two different generators (Java and Scala)

**Review by octonato — COMMENTED** [review]
Generated code still not compiling, but I believe this is a general issue in the codegen since the `scala-value-entity-customer-registry` sample is showing the same issues.

File: codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ActionServiceSourceGenerator.scala [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+      } else {
+        s"""
+           |/** Handler for "$methodName". */
+           |override def ${lowerFirst(methodName)}(${input}Src: Source[$inputTypeFullName, NotUsed]): Source[Action.Effect[$outputType], NotUsed] = {
+           |  throw new RuntimeException("The command handler for `$methodName` is not implemented, yet")
+           |}""".stripMargin
+      }
+    }
+
+    s"""|$unmanagedComment
```
Comment (patriknw) [inline codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ActionServiceSourceGenerator.scala:122]: that is added by `SourceGenerator`, can be removed here

File: codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ActionServiceSourceGenerator.scala [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+        |
+        |package $packageName
```
Comment (patriknw) [inline codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ActionServiceSourceGenerator.scala:124]: in latest ValueEntity and View generation we removed the `packageName` parameter and use `${view.fqn.parent.scalaPackage}` here

File: codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ActionServiceSourceGenerator.scala [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+      otherImports = Seq(
+        "com.akkaserverless.scalasdk.action.Action",
+        "com.akkaserverless.scalasdk.action.ActionCreationContext") ++ streamImports(service.commands),
+      semi = false)
+
+    val methods = service.commands.map { cmd =>
+      val methodName = cmd.name
+      val input = lowerFirst(cmd.inputType.name)
+      val inputTypeFullName = cmd.inputType.fullName
+      val outputType = cmd.outputType.fullName
```
Comment (patriknw) [inline codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ActionServiceSourceGenerator.scala:83]: see #448 for how to define the imports and use `typeName`

File: codegen/scala-gen/src/test/scala/com/akkaserverless/codegen/scalasdk/ActionServiceSourceGeneratorSuite.scala [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+        |  override final def additionalDescriptors: immutable.Seq[Descriptors.FileDescriptor] =
+        |    ServiceOuterClass.getDescriptor() ::
+        |    Nil
+        |
+        |  def withOptions(options: ActionOptions): MyServiceActionProvider =
+        |    new MyServiceActionProvider(actionFactory, options)
+        |}
+        |""".stripMargin)
+  }
+}
```
Comment (patriknw) [inline codegen/scala-gen/src/test/scala/com/akkaserverless/codegen/scalasdk/ActionServiceSourceGeneratorSuite.scala:224]: Scala code is looking good

**Review by patriknw — COMMENTED** [review]
looking good...

**Review by patriknw — APPROVED** [review]
LGTM, there will be some conflicts with https://github.com/lightbend/akkaserverless-java-sdk/pull/441 but I will solve those

**Pull request merged**